mongo/src/main/scala/it/agilelab/darwin/connector/mongo/MongoConnector.scala

```diff
@@ -1,19 +1,165 @@
 package it.agilelab.darwin.connector.mongo
 
+import java.util
+
+import avro.shaded.com.google.common.primitives.Bytes
+import com.mongodb.{BasicDBObject, MongoClientSettings, MongoCredential, ServerAddress}
 import com.typesafe.config.Config
-import it.agilelab.darwin.common.Connector
+import it.agilelab.darwin.common.{Connector, Logging}
 import org.apache.avro.Schema
+import com.mongodb.MongoCredential._
+import com.mongodb.client.model.Projections
+import org.apache.avro.Schema.Parser
+import org.bson.{BsonString, BsonValue}
+import org.mongodb.scala.bson.{BsonDocument, BsonString}
+import org.mongodb.scala.connection.NettyStreamFactoryFactory
+import org.mongodb.scala.{Document, FindObservable, ListCollectionsObservable, MongoClient, MongoDatabase, Observable, SingleObservable, bson}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import collection.JavaConverters._
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future}
+import org.bson.Document
+
+
+case class MongoConfig(user: String, password: String, hosts: List[String], db: String, table: String)
+
+class MongoConnector(config: Config) extends Connector with Logging {
```

Comment:
I know you aligned this implementation with the other implementation of the connectors, but maybe we should make possible to pass the connection from the outside. Maybe we can move the logic of configuration parsing to the creator. While if someone wants to initialize the connector manually, can do so by passing the connection and the other parameters.